### PR TITLE
Swarm file flavors

### PR DESCRIPTION
**Pull request by ulm0**
After reading the swarm file you guys offer, i saw it could be improved by "un-exposing" some ports for services that can talk to each other through the overlay network, and by adding some more documentation at the top of the file.

### CODE REVIEW

File: contrib/swarm/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    volumes:
+      # This directory must have cert files
+      - /var/lib/mattermost/cert:/cert:ro
+      - /etc/localtime:/etc/localtime:ro
+    environment:
+      # use app service's hostname
+      - APP_HOST=mm_app
+      # talk to the port within the overlay network
+      # without (over)exposing ports
+      - APP_PORT_NUMBER=80
```
Comment (ulm0) [inline contrib/swarm/docker-compose.yml]: Did some digging in the app container and it doesn't listen on port 8065 as stated in the current swarm file, but the port 80.

File: contrib/swarm/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      - MM_DBNAME=mattermost
+      # pass the edition to be used, default is enterprise
+      # setting this env var will make the app use the team edition
+      - edition=team
+      # in case your config is not in default location
+      # - MM_CONFIG=/mattermost/config/config.json
+    deploy:
+      restart_policy:
+        condition: on-failure
+  web:
```
Comment (ulm0) [inline contrib/swarm/docker-compose.yml]: I've also been able to make mattermost work without the web container provided here, and using traefik instead. Could add an example if wanted.

File: contrib/swarm/docker-compose.yml (hunk 1 above)
Comment (pichouk) [reply contrib/swarm/docker-compose.yml]: Good point, don't know why it was like that...

File: contrib/swarm/docker-compose.yml (hunk 2 above)
Comment (pichouk) [reply contrib/swarm/docker-compose.yml]: I'm okay to drop this container on the stack file [as we discussed here](https://github.com/mattermost/mattermost-docker/issues/203). I use (and prefer) Traefik too.

**Comment by pichouk** [comment]
Looks good to me, waiting for @xcompass approval too :)

File: contrib/swarm/docker-compose.yml (hunk 2 above)
Comment (ulm0) [reply contrib/swarm/docker-compose.yml]: As mentioned in that thread, this repo may keep flavors with the stack file, one with this web image, another with traefik.

Comment (ulm0) [reply contrib/swarm/docker-compose.yml]: As mentioned in that thread, this repo may keep flavors with the stack file, one with this web image, another with traefik.

**Comment by ulm0** [comment]
Btw i renamed the file `docker-compose.yml`. should i name it `docker-stack.yml` instead?

File: contrib/swarm/docker-compose.yml (hunk 2 above)
Comment (pichouk) [reply contrib/swarm/docker-compose.yml]: If we can keep both that's a good things, but I don't know how to do that without duplicating a lot of configuration on both files ?

**Comment by pichouk** [comment]
I prefer to keep `docker-stack.yml` unless you have a good argument to use `docker-compose.yml` ?

File: contrib/swarm/docker-compose.yml (hunk 2 above)
Comment (ulm0) [reply contrib/swarm/docker-compose.yml]: Ya, the config duplication is a problem, but I think maintaining both, the current one and the one with traefik will be enough support.

Comment (pichouk) [reply contrib/swarm/docker-compose.yml]: Ok, good for me to get both is you want to provide a Traefik example :)

**Comment by ulm0** [comment]
Nah, I just use `docker-compose.yml` for convention, since it uses the Compose file version 3 reference, although is aimed for swarm only, but i can name it back to `docker-stack.yml` in order to avoid confusion.

**Comment by ulm0** [comment]
@pichouk, just added the tfile with traefik

**Comment by pichouk** [comment]
LGTM, thanks @ulm0 :)

**Pull request merged**